### ISSUE TITLE
NN in coupling layer

### ISSUE THREAD

**Issue by sahajgarg** [chrischute/glow#2]
It seems in the paper, the description of the NN in the coupling layer is three convolutions with relu activations. Here, you also include an actnorm or batchnorm in between the convolutions. The other pytorch implementations + the tensorflow implementations don't include this. Should the normalization be there?

**Comment by chrischute** [comment]
I don't remember experimenting with this. I doubt it would make much difference, but could you try taking it out? Happy to fix it or accept a PR if you see better performance or more stable training.

**Comment by sahajgarg** [comment]
I’m going to play around more with this, it became unstable once I removed this and I’m trying to figure out why

**Comment by sahajgarg** [comment]
Turns out it depends what you want, consistency with the implementation or the paper. They do actually use actnorm in their code in the NN (different from what they describe in the paper), but slightly different from your implementation / ordering. When I change that + the initialization of the convolutions to match the tensorflow code, the model achieves lower losses more quickly (at least for the first 10 epochs). 

Also, there's some inconsistency in the paper / official implementation about preprocessing, e.g. they claim to use the preprocessing from RealNVP but don't actually perform the logit transformation. Running the model without that, it appears to perform fine.

Let me know what you'd prefer in a PR among these changes

**Comment by chrischute** [comment]
I vote for consistency with the published code because it should be able to reproduce their results. Would love a PR that matches their ordering if you saw better results over the first 10 epochs.

For the logit transformation, a flag to disable it and match their code would be great. Let's make the default flag value match their pre-processing (i.e. sounds like disabled should be default).

No worries if you don't have time to do the PR, just my two cents on what to include if you're going to make one.
